cmd/zfs/zfs_main.c

```diff
@@ -4383,6 +4383,95 @@ zfs_do_snapshot(int argc, char **argv)
 	return (-1);
 }
 
+typedef struct zfs_send_exclude_arg {
+	size_t count;
+	char **list;
+} zfs_send_exclude_arg_t;
+
+/*
+ * This function creates the zfs_send_exclude_arg_t
+ * object described above; it can be called multiple
+ * times, and the input can be comma-separated.
+ * This is NOT the most efficient data layout; however,
+ * I couldn't think of a non-pathological case where
+ * it should have more than a couple dozen instances
+ * of excludes. If that turns out to be used in
+ * practice, we might want to instead use a tree.
+ */
+static void
+add_dataset_excludes(char *in_exclude, zfs_send_exclude_arg_t *context)
+{
+	char *exclude = strdup(in_exclude);
+	size_t tempSize = context->count;
+	char **tempArray = context->list;
+	char *tempString = NULL;
+	char *exclude_orig = exclude;
+
+	for (tempString = strsep(&exclude, ",");
+	    tempString != NULL;
+	    tempString = strsep(&exclude, ",")) {
```

Comment:
isn't this, uh, `while((ts = strsep(&e, ",") != NULL)`?

Reply:
It started out as something else.  I've got it changed but I don't want to do anything since I'm trying to run the tests.